src/SymbolFinder.php

```diff
@@ -35,31 +36,65 @@ class SymbolFinder extends NodeVisitorAbstract
      */
     private $containerName;
 
+    /**
+     * @var array
+     */
+    private $nameStack = array();
+
+    /**
+     * @var array
+     */
+    private $nodeStack = array();
+
+    /**
+     * @var int
+     */
+    private $functionCount = 0;
+
     public function __construct(string $uri)
     {
         $this->uri = $uri;
     }
 
     public function enterNode(Node $node)
     {
+        array_push($this->nodeStack, $node);
+        $containerName = end($this->nameStack);
+
+        // If we enter a named node, push its name onto name stack.
+        // Else push the current name onto stack.
+        if (!empty($node->name) && (is_string($node->name) || method_exists($node->name, '__toString')) && !empty((string)$node->name)) {
+            if (empty($containerName)) {
+                array_push($this->nameStack, (string)$node->name);
+            }
+            else {
+                if ($node instanceof Node\Stmt\ClassMethod) {
+                    array_push($this->nameStack, $containerName . '::' . (string)$node->name);
```

Comment:
I don't know... probably, yes. I haven't got much experience with the PHPParser lib.